go/ephemeral/user_ek.go

```diff
@@ -246,6 +249,59 @@ func fetchUserEKStatements(ctx context.Context, g *libkb.GlobalContext, uids []k
 	return statements, nil
 }
 
+// Returns nil if the user has never published a userEK. If the user has
+// published a userEK, but has since rolled their PUK without publishing a new
+// one, this function will return wrongKID. This allows clients to chose the
+// correct generation number but not include the statement when generating a
+// new userEK.
+func fetchUserEKStatement(ctx context.Context, g *libkb.GlobalContext, uid keybase1.UID) (statement *keybase1.UserEkStatement, wrongKID bool, err error) {
+	defer g.CTrace(ctx, "fetchUserEKStatement", func() error { return err })()
+
+	apiArg := libkb.APIArg{
+		Endpoint:    "user/user_ek",
+		SessionType: libkb.APISessionTypeREQUIRED,
+		NetContext:  ctx,
+		Args: libkb.HTTPArgs{
+			"uids": libkb.S{Val: libkb.UidsToString([]keybase1.UID{uid})},
+		},
+	}
+	res, err := g.GetAPI().Get(apiArg)
+	if err != nil {
+		return nil, false, err
+	}
+
+	parsedResponse := userEKStatementResponse{}
+	err = res.Body.UnmarshalAgain(&parsedResponse)
+	if err != nil {
+		return nil, false, err
+	}
+	// User has no statements
+	if len(parsedResponse.Sigs) == 0 {
+		return nil, false, nil
+	}
+	if len(parsedResponse.Sigs) != 1 {
+		return nil, false, fmt.Errorf("Invalid server response, multiple userEK statements returned")
+	}
+	sig, ok := parsedResponse.Sigs[uid]
+	if !ok {
+		return nil, false, fmt.Errorf("Invalid server response, wrong uid returned")
+	}
+
+	statement, wrongKID, err = verifySigWithLatestPUK(ctx, g, uid, *sig)
+	// Check the wrongKID condition before checking the error, since an error
+	// is still returned in this case. TODO: Turn this warning into an error
+	// after EK support is sufficiently widespread.
+	if wrongKID {
+		g.Log.CDebugf(ctx, "It looks like you revoked a device without generating new ephemeral keys. Are you running an old version?")
+		return statement, true, nil
```

Comment:
Could we make the generation a separate parameter here? I'm a little scared that we're returning this unverified statement, and hoping that the caller doesn't trust it. It would feel better if the `statement` was `nil` in the wrong KID case.